### PR TITLE
Avalonia Demo

### PR DESCRIPTION
**Pull request by Benny121221**
**New Contributors:**
please review [CONTRIBUTING.md](https://github.com/swharden/ScottPlot/blob/master/CONTRIBUTING.md)

**Purpose:**
Gives the full demo suite and cookbook for Avalonia.

**New Functionality:**
The Avalonia demo app now has every demo that WPF has with the exception of the "Many Plots" demo.

Also apologies for the million small commits, my computer has been acting very sus recently so I committed to avoid losing data. I would normally do a few larger ones.

### CODE REVIEW

**Comment by Benny121221** [comment]
I ran code format on my machine, so I'm not sure what's going on.

**Comment by StendProg** [comment]
Hi @Benny121221, Great job!

Cookbook gives a lot of demos and requires little cost, you went even further.
Unfortunately I cannot leave a review, I am new to Avalonia.

> the million small commits

You can always squash, but in my opinion a lot of small meaningful commits are much preferable, you have it.

>  I ran code format on my machine, so I'm not sure what's going on.

Azure is much more picky than VS.
You need to add new line at the end of `src/ScottPlot.Demo.Avalonia/MessageBox.axaml ` and `src/ScottPlot.Demo.Avalonia/MessageBox.axaml.cs`. (You can find this errors on `FilesChanged` tab or on azure logs).

One question haunts me, it turns out that now you can run interactive demos on Linux, without any emulators, Mono, etc.?

**Comment by Benny121221** [comment]
> Azure is much more picky than VS.
You need to add new line at the end of src/ScottPlot.Demo.Avalonia/MessageBox.axaml and src/ScottPlot.Demo.Avalonia/MessageBox.axaml.cs. (You can find this errors on FilesChanged tab or on Azure logs).

Strange, I ran `dotnet format`... Oh well, tests pass now.

> One question haunts me, it turns out that now you can run interactive demos on Linux, without any emulators, Mono, etc.?

Yeah, I test using WSL, which you can configure a window server on through something like xMing. It actually doesn't work for a lot applications and I have no idea why. If it doesn't work for an app I normally just try a virtual machine. I don't know of any other ways but I'm sure the community has some. The Linux community are well known to be crafty.

File: src/ScottPlot.Demo.Avalonia/AvaloniaDemos/PlotViewer.axaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			</Border>
+
+		</StackPanel>
+
+		<Border>
+			<StackPanel>
+				<TextBlock Text="Sample Code"/>
+				<ScrollViewer HorizontalScrollBarVisibility="Auto">
+					<StackPanel>
+						<!-- I like Avalonia, but not allowing newlines in TextBlocks isn't quirky or cute... -->
```
Comment (FoggyFinder) [inline src/ScottPlot.Demo.Avalonia/AvaloniaDemos/PlotViewer.axaml:45]: Avalonia does support multiline TextBlock

File: src/ScottPlot.Demo.Avalonia/AboutControl.axaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+			
+			
+			<TextBlock Margin="5,10,5,5" TextWrapping="Wrap" FontSize="14">
+				This program displays ScottPlot cookbook recipes interactively.
+			</TextBlock>
+
+			<!-- Really, would it be too much to ask for linebreaks in textblocks? -->
+			
+			<TextBlock Margin="5,10,5,5" TextWrapping="Wrap" FontSize="14" Width="800" HorizontalAlignment="Left">
+				Although this program uses the WpfPlot to display plots interactively, the source code for each plot could be used in console applications and plots saved as images.
```
Comment (FoggyFinder) [inline src/ScottPlot.Demo.Avalonia/AboutControl.axaml:21]:
```suggestion
				Although this program uses the AvaPlot to display plots interactively, the source code for each plot could be used in console applications and plots saved as images.
```

File: src/ScottPlot.Demo.Avalonia/AvaloniaDemos/MouseTracker.axaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+			<Grid.ColumnDefinitions>
+				<ColumnDefinition Width="auto"/>
+				<ColumnDefinition Width="auto"/>
+				<ColumnDefinition Width="auto"/>
+			</Grid.ColumnDefinitions>
+			<Grid.RowDefinitions>
+				<RowDefinition Height="auto"/>
+				<RowDefinition Height="auto"/>
+				<RowDefinition Height="auto"/>
+			</Grid.RowDefinitions>
```
[the remark this answers is not in the document]
Comment (Benny121221) [reply src/ScottPlot.Demo.Avalonia/AvaloniaDemos/MouseTracker.axaml]: Yeah, most of these were copied straight from the WPF demos, I like the Avalonia syntax a lot more. I suspect WPF-style row/column definitions are everywhere in this project but it is what it is.

File: src/ScottPlot.Demo.Avalonia/AvaloniaDemos/PlotViewer.axaml (hunk 1 above)
Comment (Benny121221) [reply src/ScottPlot.Demo.Avalonia/AvaloniaDemos/PlotViewer.axaml]: My issue was actually the handling of tabs on newlines, not the newline characters. I was also a little too snarky, I'm sorry.

**Comment by swharden** [comment]
Hi @Benny121221, this is an impressive amount of effort. Thank you so much! Also thanks @StendProg and @FoggyFinder for the comments too. Sorry it took me so long to get to this - #605 as been absorbing a lot of my attention lately.

**Pull request merged**